src/Ast.ml

```diff
@@ -749,6 +749,14 @@ end = struct
     | ( Pat {ppat_desc= Ppat_construct _}
       , Ppat_construct ({txt= Lident "::"}, _) ) ->
         true
+    | ( (Exp {pexp_desc= Pexp_let _; _} | Str {pstr_desc= Pstr_value _; _})
+      , ( Ppat_construct (_, Some _)
+        | Ppat_variant (_, Some _)
+        | Ppat_or _ | Ppat_alias _ ) ) ->
+        (* let (Some p) = e *)
+        (* let (P1 | P2) = e *)
+        (* let (pat as ident) = e *)
```

Comment:
Nit: Do you think that it is valuable to have only a few such comments indicating the concrete syntax corresponding to the parsetree constructors? I was basically assuming that anyone reading to code would either be somewhat familiar with the parsetree, or willing to read it as they went.

Reply:
It was only meaningful internally to help understand what diffs can be upstream. Now removed.